R/dgp-lib.R

```diff
@@ -529,7 +531,9 @@ correlated_logistic_gaussian_dgp <- function(n, p_uncorr, p_corr,
 lss_gaussian_dgp <- function(n, p, k, s, thresholds = 0, signs = 1, betas = 1,
                              intercept = 0, overlap = FALSE, err = NULL,
                              data_split = FALSE, train_prop = 0.5,
-                             return_values = c("X", "y", "support"), ...) {
+                             return_values = c("X", "y", "support",
```

Comment:
I may have missed this, but we should be sure to document what each of these mean (esp. "support" and "int_support"). 

I'm also not sure that "support" is the right term here. When I think of "support" I think of the set of possible values that a random variable can take. Maybe "active_features_index"?

Reply:
I have currently documented these in `generate_y_*()`, but I'll can copy this over to the `*_dgp_fun()` documentation too.

I was using the term "support" as it's used in the [sparsistency](https://en.wiktionary.org/wiki/sparsistency) literature. I think this is common in statistics but am open to other terms. I just worry "active_features_index" is too long.

As an aside, I think we should eventually return the value of the feature coefficients, rather than just the indices of the active support. This could impact how we decide to return this feature-level information...

Reply:
I see! I wasn't familiar with that usage of "support", but I think it could still be a bit confusing since random variables are involved here. Maybe just changing to "betas_support" would work.

Re. issue #110, I agree that's a good idea and could also help with the naming, e.g. we could use "beta_nonzero" as one of the return values, which could be some sort of data structure holding the coefficients and their indices at the same time. The package Matrix has sparse vectors, though I hesitate to add another dependency to our package. But it may be worth it if we have other uses for it (e.g. using sparse matrices). Otherwise, maybe we just create a simple S3 class that replicates `sparseVector`.